jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh

```diff
@@ -157,31 +157,44 @@ function configure_EJB_Timer_datasource {
     fi
 }
 
-# Sets the NONXA url if only the PREFIX_XA_CONNECTION_PROPERTY_URL is provided and vice-versa.
+# finds the URL, precedence is XA_CONNECTION_PROPERTY_Url > XA_CONNECTION_PROPERTY_URL > URL
+# $1 - datasource prefix
+function find_url() {
+    local url=$(find_env "${1}_URL")
+    local xaUrl=$(find_env "${1}_XA_CONNECTION_PROPERTY_Url")
+    if [ ! -z "${xaUrl}" ]; then
+        echo ${xaUrl}
+    else
+      url=$(find_env "${1}_XA_CONNECTION_PROPERTY_URL" "${url}")
+      echo ${url}
+    fi
+}
+
+# Sets the NONXA url if only the PREFIX_XA_CONNECTION_PROPERTY_URL|Url is provided and vice-versa.
 # $1 - datasource prefix
 function set_url {
-    local prefixedUrl=$(find_env "${1}_URL")
-    url=$(find_env "${1}_XA_CONNECTION_PROPERTY_URL" "${prefixedUrl}")
+    local url=$(find_url "${1}")
     # remove spaces, newlines, etc from url, see RHPAM-3808
     url=$(echo ${url} | tr -d '[:space:]')
-    if [ "${prefixedUrl}x" = "x" ]; then
-        # KIECLOUD-598 We need to escape the & if it isn't already escaped...
-        url=$(echo "${url}" | sed 's|\([^\\]\)&amp|\1\\\&amp|g')
-        # KIECLOUD-598 ...and we need to escape also the ; only in this case
-        url=$(echo ${url} | sed -e 's/\;/\\;/g')
-        eval ${1}_URL="${url}"
+
+    # KIECLOUD-598 We need to escape the & if it isn't already escaped...
+    url=$(echo "${url}" | sed 's/&/\\&amp;/g')
```

Comment:
Don't we need to verify that & is not already escaped? And so that there isn't already before \\ ?
See line 178

Reply:
that's a good question, this func is called just once and prepare the provided value from user, at this point, we don't expect user escaping these special characters.

but it can be a valid point to prevent issues if user escapes it.

Reply:
updated.